ImageGallery/src/org/sleuthkit/autopsy/imagegallery/datamodel/grouping/GroupManager.java

```diff
@@ -107,6 +107,11 @@
 
     private final ImageGalleryController controller;
 
+    /**
+     * Keeps track of the last path group
+     *  - a change in path indicates the last path group is analyzed
+     */
+    private GroupKey<?> lastUpdatedPathGroup = null;
```

Comment:
Is it possible that this member variable may be accessed by multiple threads?

Reply:
Ugh, yes.  Thanks for catching this, I'll fix it.

Reply:
I see that there is now an annotation indicating that the intrinsic lock (monitor) of the GroupManager is used to control access to lastUpdatedPathGroup. However, the access in checkForPathGroupChange is only guarded indirectly by having the sole call to it happening from a synchronized method. It would be safer and more clear to synchronize checkForPathGroupChange as well. Also, there is no guarding at all in resetLastUpdatedPathGroup, which is not called from a synchronized method.